src/bin/pgcopydb/dump_restore.c

```diff
@@ -622,169 +786,48 @@ copydb_write_restore_list(CopyDataSpec *specs, PostgresDumpSection section)
-		{
-			skip = true;
-			log_notice("Skipping COMMENT ON EXTENSION \"%s\"", name);
-		}
-
-		if (!skip && catOid == PG_NAMESPACE_OID)
-		{
-			bool exists = false;
-
-			if (!copydb_schema_already_exists(specs, oid, &exists))
-			{
-				log_error("Failed to check if restore name \"%s\" "
-						  "already exists",
-						  name);
-				destroyPQExpBuffer(listContents);
-				return false;
-			}
-
-			if (exists)
-			{
-				skip = true;
-
-				log_notice("Skipping already existing dumpId %d: %s %u %s",
-						   contents.array[i].dumpId,
-						   contents.array[i].description,
-						   contents.array[i].objectOid,
-						   contents.array[i].restoreListName);
-			}
-		}
-
-		if (!skip && copydb_objectid_has_been_processed_already(specs, item))
-		{
-			skip = true;
-
-			log_notice("Skipping already processed dumpId %d: %s %u %s",
-					   contents.array[i].dumpId,
-					   contents.array[i].description,
-					   contents.array[i].objectOid,
-					   contents.array[i].restoreListName);
-		}
-
-		/*
-		 * For SEQUENCE catalog entries, we want to limit the scope of the hash
-		 * table search to the OID, and bypass searching by restore name. We
-		 * only use the restore name for the SEQUENCE OWNED BY statements.
-		 *
-		 * This also allows complex filtering of sequences that are owned by
-		 * table a and used as a default value in table b, where table a has
-		 * been filtered-out from pgcopydb scope of operations, but not table
-		 * b.
-		 */
-		if (item->desc == ARCHIVE_TAG_SEQUENCE)
-		{
-			name = NULL;
-		}
-
-		/*
-		 * There could be a case where the materalized view is included in the
-		 * dump, but the refresh is filtered out using [exclude-table-data].
-		 * In this case, we want to skip the refresh.
-		 */
-		if (!skip &&
-			item->desc == ARCHIVE_TAG_REFRESH_MATERIALIZED_VIEW &&
-			copydb_matview_refresh_is_filtered_out(specs, oid))
-		{
-			skip = true;
-
-			log_notice("Skipping materialized view refresh dumpId %d: %s %u %s",
-					   contents.array[i].dumpId,
-					   contents.array[i].description,
-					   contents.array[i].objectOid,
-					   contents.array[i].restoreListName);
-		}
-
-		if (!skip && copydb_objectid_is_filtered_out(specs, oid, name))
-		{
-			skip = true;
-
-			log_notice("Skipping filtered-out dumpId %d: %s %u %u %s",
-					   contents.array[i].dumpId,
-					   contents.array[i].description,
-					   contents.array[i].catalogOid,
-					   contents.array[i].objectOid,
-					   contents.array[i].restoreListName);
-		}
-
-		appendPQExpBuffer(listContents, "%s%d; %u %u %s %s\n",
-						  skip ? ";" : "",
-						  contents.array[i].dumpId,
-						  contents.array[i].catalogOid,
-						  contents.array[i].objectOid,
-						  contents.array[i].description,
-						  contents.array[i].restoreListName);
+		log_error("Failed to open list file \"%s\": %m", listFilename);
+		return false;
 	}
+	ArchiveContext context = { 0 };
+	context.specs = specs;
+	context.line = line;
+	context.listFile = listFile;
```

Comment:
Please embed the structure slots definitions within the static definition, as it's all pointers anyways, like we do in all the rest of the code.